compiler/rustc_passes/src/check_const.rs

```diff
@@ -59,12 +60,70 @@ impl NonConstExpr {
 fn check_mod_const_bodies(tcx: TyCtxt<'_>, module_def_id: LocalDefId) {
     let mut vis = CheckConstVisitor::new(tcx);
     tcx.hir().visit_item_likes_in_module(module_def_id, &mut vis.as_deep_visitor());
+    if tcx.features().enabled(sym::const_trait_impl) {
```

Comment:
I think you can skip this check. We can emit these errors even when the feature gate is turned off.